addons/sale_project/models/sale_order.py

```diff
@@ -138,6 +138,16 @@ def _action_confirm(self):
             # Orders from different companies are confirmed together
             for order in self:
                 order.order_line.sudo().with_company(order.company_id)._timesheet_service_generation()
+
+        # If the order has exactly one project and that project comes from a template, set the company of the template
+        # on the project.
+        for order in self:
+            if len(order.project_ids) == 1:
+                project = order.project_ids[0]
+                for sol in order.order_line:
+                    if project == sol.project_id and (project_template := sol.product_template_id.project_template_id):
+                        project.sudo().company_id = project_template.company_id
```

Comment:
the project template could also be private, no? If yes, then I guess we should also add a sudo to read the company. 🙂 

```suggestion
                        project.sudo().company_id = project_template.sudo().company_id
```

Reply:
You are right, it could be :+1: